libgadget/drift.c

```diff
@@ -138,21 +138,21 @@ static void real_drift_particle(int i, inttime_t ti1, const double ddrift, const
 }
 
 /* Update all particles to the current time, shifting them by a random vector.*/
-void drift_all_particles(inttime_t ti1, const double random_shift[3])
+void drift_all_particles(inttime_t ti1, const double BoxSize, Cosmology * CP, const double random_shift[3])
```

Comment:
What about combinding BoxSize, CP, random_shift into an object? e.g. TimeIntegrator?

Reply:
This suggests a better idea: move CP inside PartManager

Reply:
I might move BoxSize in there as well next time and then make the petaio functions take a single argument which is a struct part_manager

Reply:
The name TimeIntegrator can smoothly related to the behaviours -- a time integrator drifts and kicks particles; so it's a reusable interface instead of a bare collection of plain old data. 

Combining interfaces into PartManager is producing a bigger interface, which usually feel somewhat uncomfortable. But rules are there to be broken..

Reply:
I'm sorry, I meant random_shift, not CP! random_shift is not really a property of the time integration: it is a property of the particle table and as such belongs in the PartManager.

Reply:
This does also suggest moving Time et al into CP - maybe in a future pull.